src/index/DeltaTriples.cpp

```diff
@@ -237,31 +240,37 @@ ReturnType DeltaTriplesManager::modify(
         };
         auto writeAndUpdateSnapshot = [&updateSnapshot, &deltaTriples,
                                        writeToDiskAfterRequest]() {
+          ad_utility::Timer timer{ad_utility::Timer::InitialStatus::Started};
+          DeltaTriplesModifyTimings timings;
           if (writeToDiskAfterRequest) {
-            // TODO<RobinTF> Find a good way to track the time it takes to write
-            // this and store it into the corresponding metadata object.
             deltaTriples.writeToDisk();
+            timings.diskWritebackTime_ = timer.msecs();
+            timer.start();
           }
           updateSnapshot();
+          timings.snapshotUpdateTime_ = timer.msecs();
+          return timings;
         };
 
         if constexpr (std::is_void_v<ReturnType>) {
           function(deltaTriples);
-          writeAndUpdateSnapshot();
+          return writeAndUpdateSnapshot();
         } else {
           ReturnType returnValue = function(deltaTriples);
-          writeAndUpdateSnapshot();
-          return returnValue;
+          auto timings = writeAndUpdateSnapshot();
+          return std::make_tuple(returnValue, timings);
         }
       });
 }
-// Explicit instantions
-template void DeltaTriplesManager::modify<void>(
+// Explicit instantiations
+template DeltaTriplesModifyTimings DeltaTriplesManager::modify<void>(
     std::function<void(DeltaTriples&)> const&, bool writeToDiskAfterRequest);
```

Comment:
`template Timings DeltaTriplesManager::modify<void>(Func<void> const&, bool);` (with appropriate (templated) alias template `F` and alias `Timings`.

Alternatviely use a very local macro.
```
INSTANTIATE_DELTA_TRIPLES_MODIFY(void);
INSTANTIATE_DELTA_TRIPLES_MODIFY(int);
```